pandas/core/algorithms.py

```diff
@@ -2203,3 +2203,31 @@ def _sort_tuples(values: np.ndarray[tuple]):
     arrays, _ = to_arrays(values, None)
     indexer = lexsort_indexer(arrays, orders=True)
     return values[indexer]
+
+
+def re_sort_union_after_inputs(union_values, lvals, rvals) -> np.ndarray:
```

Comment:
why is this the result? why isn't it just a regular sort?

Reply:
We want to achieve the following:

```
idx1 = Index([1, 0, 0])
idx2 = Index([1, 0])
idx1.union(idx2, sort=False)
```
The result of 
```
result = self._outer_indexer(np.sort(lvals), np.sort(rvals))[0]
```
is 
```
np.array([0, 0, 1])
```
Which has the wrong order, so we have to re-sort after the input order, which is what the function is doing. It returns
```
np.array([1, 0, 0])
```

Reply:
why is the 2nd one correct?

Reply:
Union should not sort if it is set to False? So we should keep the input Order?